### PR TITLE
601 Create standalone SNP Track

### PR DESCRIPTION
**Pull request by peterkxie**
Creates new SNP Track that uses a SNPAdapter and SNPXYRenderer. SNPAdapter wraps either a Bam or Cram Adapter using a 'subadapter' concept (thanks to @cmdcolin) and draws mismatches onto a canvas using coverage bin information. Still working on some deeper integration testing.

Issue with using types: 'frozen' left in for now, idea of changing to pluginManager.pluggableConfigSchemaType('adapter') commented out for now due to difficulty of getting pluginManager to work in an Adapter's index.js file. Since there are two adapters (the main SNP adapter and one sub adapter), has issues identifying the correct adapter for config schema

### CODE REVIEW

File: packages/alignments/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{
+  "name": "@gmod/jbrowse-plugin-alignments",
```
Comment (cmdcolin) [inline packages/alignments/package-lock.json]: We don't want to keep package-lock.json files...may want to create a gitignore for it also if this was accidentally committed

File: packages/alignments/src/BamAdapter/BamAdapter.test.js [hunk 2]
```diff
@@ -57,3 +57,50 @@ test('test usage of BamSlightlyLazyFeature toJSON (used in the drawer widget)',
...
+        bpPerPx: 0.2,
+      },
+    },
+    100,
+  )
+
+  expect(stats).toEqual(
+    expect.objectContaining({
+      scoreMin: 0,
+      scoreMax: expect.any(Number),
```
Comment (cmdcolin) [inline packages/alignments/src/BamAdapter/BamAdapter.test.js]: Can you make the scoreMax the actual value that it should be e.g. the actual maximum value over this range?

File: packages/alignments/package-lock.json (hunk 1 above)
Comment (peterkxie) [reply packages/alignments/package-lock.json]: yeah I think that was commited on accident during one of my pushes

File: packages/alignments/src/CramAdapter/CramAdapter.test.js [hunk 3]
```diff
@@ -128,3 +128,47 @@ test('test usage of cramSlightlyLazyFeature toJSON (used in the drawer widget)',
...
+        bpPerPx: 0.2,
+      },
+    },
+    100,
+  )
+
+  expect(stats).toEqual(
+    expect.objectContaining({
+      scoreMin: 0,
+      scoreMax: expect.any(Number),
```
Comment (cmdcolin) [inline packages/alignments/src/CramAdapter/CramAdapter.test.js]: Same actual scoreMax here

**Comment by cmdcolin** [comment]
Can we possibly rename this from "SNP track"? In JBrowse 1 it is called SNPCoverage. The "SNP track" is not really descriptive, the main function is the coverage really

File: packages/alignments/package-lock.json (hunk 1 above)
Comment (cmdcolin) [reply packages/alignments/package-lock.json]: no worries

**Comment by peterkxie** [comment]
I was following the track naming convention since it's a track type, should I try renaming to SNPCoverageTrack?

**Comment by cmdcolin** [comment]
@peterkxie  That would be my preference, yea

File: test_data/config_integration_test.json [hunk 4]
```diff
@@ -528,6 +528,9 @@
       "type": "SNPCoverageTrack",
       "trackId": "volvox_alignments_bam_nonexist_SNP",
       "name": "volvox-sorted red/blue (bam nonexist 404) SNP",
+      "assemblyNames": [
+        "volvox"
+      ],
```
Comment (cmdcolin) [inline test_data/config_integration_test.json]: :+1:

**Comment by cmdcolin** [comment]
One thing I noticed. Currently if you have a deletion represented by a literal D in the CIGAR for example, we represent this as a different color in the SNP coverage. I think that to be fully accurate, we should really represent this as a drop in coverage.

This is something that jbrowse 1 may not be currently handling properly either

**Comment by cmdcolin** [comment]
I pushed a change that makes it so deletion does not count towards coverage, here is how it can look

t11 png

You can see there are no reads covering this area, even though there are reads with DEL spanning the gap

**Comment by cmdcolin** [comment]
I think this is probably good to go to me. Only loose ends perhaps that are notable are

1) the wiggle tooltips could maybe use some work, as the snpcoverage logic is embedded in the wigglerendering, and maybe I would change the styling possibly
2) the types.frozen remains kind of weird but seems essential as we could not recursively refer to subadapter types from what I could tell. An alternative is that perhaps the subadapter is not a child of the adapter

**Comment by rbuels** [comment]
need to test that you can add one of these tracks using the + button in the bottom of the track selector

**Comment by cmdcolin** [comment]
This seems like it should be good to go since we adding the add track procedure works

**Comment by cmdcolin** [comment]
:shipit:

**Comment by cmdcolin** [comment]
image

**Pull request merged**